src/conn/conn_sweep.c

```diff
@@ -45,6 +80,21 @@ __sweep_mark(WT_SESSION_IMPL *session, uint64_t now)
           __wt_atomic_loadi32(&dhandle->session_inuse) > 0 || dhandle->timeofdeath != 0)
             continue;
 
+        /* For table dhandles, skip expiration if associated file dhandles exist. */
+        if (dhandle->type == WT_DHANDLE_TYPE_TABLE) {
+            WT_WITH_TABLE_READ_LOCK(session,
+              WT_WITHOUT_DHANDLE(session,
```

Comment:
I think that you do not need the `WT_WITHOUT_DHANDLE` macro here.

Reply:
@kommiharibabu Fixed in 0e6cf0b3b55c1e708afae229b5894b26c41eb1a0